core-web/libs/portlets/edit-ema/portlet/src/lib/components/dot-ema-dialog/store/dot-ema-dialog.store.ts

```diff
@@ -0,0 +1,190 @@
+import { ActionPayload } from '../../../shared/models';
+
+type dialogType = 'content' | 'form' | 'widget' | null;
+
+export interface EditEmaDialogState {
+    visible: boolean;
+    header: string;
+    loading: boolean;
+    url: string;
+    type: dialogType;
+    payload?: ActionPayload;
+}
+
+@Injectable()
+export class DotEmaDialogStore extends ComponentStore<EditEmaDialogState> {
+    constructor() {
+        super({ header: '', loading: false, url: '', type: null, visible: false });
+    }
+
+    private dotActionUrl = inject(DotActionUrlService);
+
+    readonly dialogState$ = this.select((state) => state);
+
+    /**
+     * Create a contentlet from the palette
+     *
+     * @memberof EditEmaStore
+     */
+    readonly createContentletFromPalette = this.effect(
+        (
+            contentTypeVariable$: Observable<{
+                variable: string;
+                name: string;
+                payload: ActionPayload;
+            }>
+        ) => {
+            return contentTypeVariable$.pipe(
+                switchMap(({ name, variable, payload }) => {
+                    return this.dotActionUrl.getCreateContentletUrl(variable).pipe(
+                        tapResponse(
+                            (url) => {
+                                this.openCreateIframe({
+                                    url,
+                                    contentType: name,
+                                    payload
+                                });
+                            },
+                            (e) => {
+                                console.error(e);
+                            }
+                        )
+                    );
+                })
+            );
+        }
+    );
+
+    // This method is called when the user clicks in the + button in the jsp dialog or drag a contentType from the palette
+    readonly openCreateIframe = this.updater(
+        (
+            state,
+            {
+                url,
+                contentType,
+                payload
+            }: { url: string; contentType: string; payload?: ActionPayload }
+        ) => {
+            return {
+                ...state,
+                url: url,
+                header: `Create ${contentType}`,
+                loading: true,
+                type: 'content',
+                visible: true,
+                payload
+            };
+        }
+    );
+
+    // This method is called when the user clicks on the edit button
+    readonly openEditIframe = this.updater((state, payload: { inode: string; title: string }) => {
+        return {
+            ...state,
+            header: payload.title,
+            loading: true,
+            url: this.createEditContentletUrl(payload.inode),
+            type: 'content',
+            visible: true
+        };
+    });
+
+    // This method is called when the user clicks on the [+ add] button
+    readonly openAddIframe = this.updater(
```

Comment:
```suggestion
    readonly addContent = this.updater(
```

The fact that the to add a content lives in an iframe should not drive the name of the updater, if tomorrow we change the iframe for the new edit content the name will stop making sense.